### PR TITLE
Support for opentera online

### PR DESCRIPTION
**Pull request by philippewarren**

### CODE REVIEW

**Comment by philippewarren** [comment]
Draft while waiting on `opentera_webrtc_ros` being merged

File: ros/demos/ttop_opentera/launch/ttop_opentera.launch [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <launch>
     <include file="$(find t_top)/launch/opentera_teleop.launch">
         <arg name="visualization" value="true"/>
+        <arg name="is_stand_alone" value="true"/>
```
Comment (mamaheux) [inline ros/demos/ttop_opentera/launch/ttop_opentera.launch:4]: The argument has been removed.

File: ros/t_top/launch/behaviors/opentera.launch [hunk 2]
```diff
@@ -3,16 +3,22 @@
     <arg name="signaling_server_port" default="8080"/>
     <arg name="signaling_server_password" default="abc"/>
     <arg name="database_path" default="~/.ros/opentera.db"/>
+    <arg name="is_stand_alone" default="false"/>
```
Comment (mamaheux) [inline ros/t_top/launch/behaviors/opentera.launch:6]: The argument has been removed.

File: ros/demos/ttop_opentera/launch/ttop_opentera.launch (hunk 1 above)
Comment (philippewarren) [reply ros/demos/ttop_opentera/launch/ttop_opentera.launch:5]: Not sure what you mean

File: ros/t_top/launch/behaviors/opentera.launch (hunk 2 above)
Comment (philippewarren) [reply ros/t_top/launch/behaviors/opentera.launch:5]: Not sure what you mean

File: ros/t_top/scripts/robot_status.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        rospy.init_node("robot_status_publisher")
+
+        self.status_pub = rospy.Publisher(
+            '/robot_status', RobotStatus, queue_size=10)
+        self.status_webrtc_pub = rospy.Publisher(
+            '/webrtc_data_outgoing', String, queue_size=10)
+
+        self.battery_status_sub = rospy.Subscriber(
+            "/opencr/base_status",
+            Float32MultiArray, self.battery_status_cb, queue_size=1)
```
Comment (mamaheux) [inline ros/t_top/scripts/robot_status.py:50]: Rename to base_status_sub

File: ros/t_top/scripts/robot_status.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        while not rospy.is_shutdown():
+            # Fill timestamp
+            status = RobotStatus()
+            status.header.stamp = rospy.Time.now()
+
+            # Fill robot info
+            status.battery_voltage = self.battery_status.voltage
+            status.battery_current = self.battery_status.current
+            status.battery_level = self.battery_status.percentage
+            status.is_charging = self.battery_status.is_plugged_in
```
Comment (mamaheux) [inline ros/t_top/scripts/robot_status.py:85]: We can have a race condition here. I would add a lock here and in battery_status_cb.

File: ros/t_top/scripts/robot_status.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        result = os.statvfs(mount_point)
+        total_blocks = result.f_blocks
+        free_blocks = result.f_bfree
+        return 100 - (free_blocks * 100 / total_blocks)
+
+    def battery_status_cb(self, msg):
+        self.battery_status = BatteryStatus(*msg.data)
+
+    def run(self):
+        r = rospy.Rate(self.pub_rate)
```
Comment (mamaheux) [inline ros/t_top/scripts/robot_status.py:75]: Rename to rate

**Review by doumdi — APPROVED** [review]

**Review by mamaheux — APPROVED** [review]
You can merge after adding information in the readme.

File: tools/ca_certificate_setup.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# Generating CA Certificates for the robot
```
Comment (mamaheux) [inline tools/ca_certificate_setup.md:1]: Can you create a folder for `ca_certificate_setup.md`, `fetch_certificates.sh` and `generate_certificate.sh`?

File: tools/ca_certificate_setup.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+## 1. Generate CA Certificates on the robot
```
Comment (mamaheux) [inline tools/ca_certificate_setup.md:3]: Can you refer this readme in the teleop demo one?

**Review by mamaheux — APPROVED** [review]

File: tools/ca_certificate_setup.md (hunk 7 above)
Comment (philippewarren) [reply tools/ca_certificate_setup.md:3]: Already doing so here: https://github.com/introlab/t-top/pull/27/files#diff-12e4318ecc06adce5464053621d818045a1d90a1f8a3a0f94f4646b489be343aR15

**Pull request merged**